<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>DSP Analysis QC Report</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="DSP_Analysis_QC_Report_files/libs/clipboard/clipboard.min.js"></script>
<script src="DSP_Analysis_QC_Report_files/libs/quarto-html/quarto.js"></script>
<script src="DSP_Analysis_QC_Report_files/libs/quarto-html/popper.min.js"></script>
<script src="DSP_Analysis_QC_Report_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="DSP_Analysis_QC_Report_files/libs/quarto-html/anchor.min.js"></script>
<link href="DSP_Analysis_QC_Report_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="DSP_Analysis_QC_Report_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="DSP_Analysis_QC_Report_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="DSP_Analysis_QC_Report_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="DSP_Analysis_QC_Report_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">DSP Analysis QC Report</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="load-libraries" class="level2">
<h2 class="anchored" data-anchor-id="load-libraries">Load Libraries</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load all relevant libraries</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DSPWorkflow)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GeomxTools)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggforce)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(shadowtext)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(PCAtools)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gridExtra)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(grid)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gt)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(openxlsx)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ComplexUpset)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"DSP_QC_functions.R"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="initialization" class="level2">
<h2 class="anchored" data-anchor-id="initialization">Initialization</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Input file parameters</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>pkc.file.name <span class="ot">&lt;-</span> <span class="st">"Hs_R_NGS_WTA_v1.0.pkc"</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>pkc.file.path <span class="ot">&lt;-</span> <span class="fu">paste0</span>(params<span class="sc">$</span>data.folder, pkc.file.name)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>annotation.file.name <span class="ot">&lt;-</span> <span class="st">"annotation_Krug_CPTR_5_april2024_NC_edit.xlsx"</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>annotation.file.path <span class="ot">&lt;-</span> <span class="fu">paste0</span>(params<span class="sc">$</span>data.folder, annotation.file.name)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>dcc.files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="fu">file.path</span>(<span class="fu">paste0</span>(params<span class="sc">$</span>data.folder, <span class="st">"dcc"</span>)),</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">pattern =</span> <span class="st">".dcc$"</span>,</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">full.names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">recursive =</span> <span class="cn">TRUE</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Annotation parameters</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>annotation.sheet.name <span class="ot">&lt;-</span> <span class="st">"annotation"</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>sample.id.field.name <span class="ot">&lt;-</span> <span class="st">"Sample_ID"</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>roi.field.name <span class="ot">&lt;-</span> <span class="st">"ROI"</span></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>panel.field.name <span class="ot">&lt;-</span> <span class="st">"panel"</span></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>slide.field.name <span class="ot">&lt;-</span> <span class="st">"slide name"</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>class.field.name <span class="ot">&lt;-</span> <span class="st">"class"</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>region.field.name <span class="ot">&lt;-</span> <span class="st">"Region"</span></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>segment.field.name <span class="ot">&lt;-</span> <span class="st">"segment"</span></span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>area.field.name <span class="ot">&lt;-</span> <span class="st">"area"</span></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>nuclei.field.name <span class="ot">&lt;-</span> <span class="st">"nuclei"</span></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>exclude.sankey <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>segment.id.length <span class="ot">&lt;-</span> <span class="dv">8</span></span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the GeoMxSet Object</span></span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>init.object <span class="ot">&lt;-</span> <span class="fu">initialize_object</span>(<span class="at">dcc.files =</span> dcc.files,</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>                            <span class="at">pkc.files =</span> pkc.file.path,</span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>                            <span class="at">annotation.file =</span> annotation.file.path,</span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>                            <span class="at">annotation.sheet.name =</span> annotation.sheet.name,</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>                            <span class="at">sample.id.field.name =</span> sample.id.field.name,</span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>                            <span class="at">roi.field.name =</span> roi.field.name,</span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>                            <span class="at">panel.field.name =</span> panel.field.name,</span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>                            <span class="at">slide.field.name =</span> slide.field.name, </span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a>                            <span class="at">class.field.name =</span> class.field.name, </span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a>                            <span class="at">region.field.name =</span> region.field.name, </span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a>                            <span class="at">segment.field.name =</span> segment.field.name,</span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a>                            <span class="at">area.field.name =</span> area.field.name,</span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a>                            <span class="at">nuclei.field.name =</span> nuclei.field.name, </span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a>                            <span class="at">segment.id.length =</span> segment.id.length)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in readNanoStringGeoMxSet(dccFiles = dcc.files, pkcFiles = pkc.files, :
Annotations missing for the following: DSP-1001660014796-H-A04.dcc,
DSP-1001660014796-H-A10.dcc, DSP-1001660014796-H-B04.dcc,
DSP-1001660014796-H-E05.dcc, DSP-1001660014796-H-E11.dcc,
DSP-1001660014797-G-E02.dcc, DSP-1001660014797-G-E03.dcc, These will be
excluded from the GeoMxSet object.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in readNanoStringGeoMxSet(dccFiles = dcc.files, pkcFiles = pkc.files, :
Annotations missing for the following: DSP-1001660014796-H-A04.dcc,
DSP-1001660014796-H-A10.dcc, DSP-1001660014796-H-B04.dcc,
DSP-1001660014796-H-E05.dcc, DSP-1001660014796-H-E11.dcc,
DSP-1001660014797-G-E02.dcc, DSP-1001660014797-G-E03.dcc, These will be
excluded from the GeoMxSet object.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in initialize_object(dcc.files = dcc.files, pkc.files = pkc.file.path, : area is not found in the annotation and will not be considered </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in initialize_object(dcc.files = dcc.files, pkc.files = pkc.file.path, : nuclei is not found in the annotation and will not be considered </code></pre>
</div>
</div>
</section>
<section id="object-summary" class="level2">
<h2 class="anchored" data-anchor-id="object-summary">Object Summary</h2>
<p><a href="#fig-sankey">Figure&nbsp;1</a> shows a summary of AOIs per annotation</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Rename the slide name column for formatting</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pData</span>(init.object) <span class="ot">&lt;-</span> <span class="fu">pData</span>(init.object) <span class="sc">%&gt;%</span> </span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">slide =</span> <span class="fu">gsub</span>(<span class="st">"slide_"</span>, <span class="st">""</span>, slide_name))</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the lanes of the Sankey plot</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>lane1 <span class="ot">&lt;-</span> <span class="st">"slide"</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>lane2 <span class="ot">&lt;-</span> <span class="st">"class"</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>lane3 <span class="ot">&lt;-</span> <span class="st">"region"</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>lane4 <span class="ot">&lt;-</span> <span class="st">"segment"</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>fill_lane <span class="ot">&lt;-</span> <span class="st">"region"</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>lanes <span class="ot">&lt;-</span> <span class="fu">c</span>(lane1, lane2, lane3, lane4)</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="co">#Establish variables for the Sankey plot</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> id <span class="ot">&lt;-</span> y <span class="ot">&lt;-</span> n <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="co"># select the annotations we want to show, use `` to surround column</span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a><span class="co"># names with spaces or special symbols</span></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a count matrix</span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>count.mat <span class="ot">&lt;-</span> <span class="fu">count</span>(<span class="fu">pData</span>(init.object), </span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>                   <span class="sc">!!</span><span class="fu">as.name</span>(lane1), </span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>                   <span class="sc">!!</span><span class="fu">as.name</span>(lane2), </span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>                   <span class="sc">!!</span><span class="fu">as.name</span>(lane3), </span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>                   <span class="sc">!!</span><span class="fu">as.name</span>(lane4))</span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove any rows with NA values</span></span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>na.per.column <span class="ot">&lt;-</span> <span class="fu">colSums</span>(<span class="fu">is.na</span>(count.mat))</span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a>na.total.count <span class="ot">&lt;-</span> <span class="fu">sum</span>(na.per.column)</span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a>                                             </span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(na.total.count <span class="sc">&gt;</span> <span class="dv">0</span>){</span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a>  count.mat <span class="ot">&lt;-</span> count.mat[<span class="sc">!</span><span class="fu">rowSums</span>(<span class="fu">is.na</span>(count.mat)),]</span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(count.mat) <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(count.mat)</span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Gather the data and plot in order: lane 1, lane 2, ..., lane n</span></span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a><span class="co"># gather_set_data creates x, id, y, and n fields within sankey.count.data</span></span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Establish the levels of the Sankey</span></span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a>sankey.count.data <span class="ot">&lt;-</span> <span class="fu">gather_set_data</span>(count.mat, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)</span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the annotations to use for the Sankey x axis labels</span></span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a>sankey.count.data<span class="sc">$</span>x[sankey.count.data<span class="sc">$</span>x <span class="sc">==</span> <span class="dv">1</span>] <span class="ot">&lt;-</span> lane1</span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a>sankey.count.data<span class="sc">$</span>x[sankey.count.data<span class="sc">$</span>x <span class="sc">==</span> <span class="dv">2</span>] <span class="ot">&lt;-</span> lane2</span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a>sankey.count.data<span class="sc">$</span>x[sankey.count.data<span class="sc">$</span>x <span class="sc">==</span> <span class="dv">3</span>] <span class="ot">&lt;-</span> lane3</span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a>sankey.count.data<span class="sc">$</span>x[sankey.count.data<span class="sc">$</span>x <span class="sc">==</span> <span class="dv">4</span>] <span class="ot">&lt;-</span> lane4</span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a>sankey.count.data<span class="sc">$</span>x <span class="ot">&lt;-</span></span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a>    <span class="fu">factor</span>(</span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a>      sankey.count.data<span class="sc">$</span>x,</span>
<span id="cb7-52"><a href="#cb7-52" aria-hidden="true" tabindex="-1"></a>      <span class="at">levels =</span> <span class="fu">c</span>(<span class="fu">as.name</span>(lane1), </span>
<span id="cb7-53"><a href="#cb7-53" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">as.name</span>(lane2), </span>
<span id="cb7-54"><a href="#cb7-54" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">as.name</span>(lane3), </span>
<span id="cb7-55"><a href="#cb7-55" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">as.name</span>(lane4)))</span>
<span id="cb7-56"><a href="#cb7-56" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb7-57"><a href="#cb7-57" aria-hidden="true" tabindex="-1"></a><span class="co"># For position of Sankey 100 segment scale</span></span>
<span id="cb7-58"><a href="#cb7-58" aria-hidden="true" tabindex="-1"></a>adjust.scale.pos <span class="ot">=</span> <span class="sc">-</span><span class="fl">1.1</span></span>
<span id="cb7-59"><a href="#cb7-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-60"><a href="#cb7-60" aria-hidden="true" tabindex="-1"></a><span class="co"># plot Sankey diagram</span></span>
<span id="cb7-61"><a href="#cb7-61" aria-hidden="true" tabindex="-1"></a>sankey.plot <span class="ot">&lt;-</span></span>
<span id="cb7-62"><a href="#cb7-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(sankey.count.data,</span>
<span id="cb7-63"><a href="#cb7-63" aria-hidden="true" tabindex="-1"></a>         <span class="fu">aes</span>(</span>
<span id="cb7-64"><a href="#cb7-64" aria-hidden="true" tabindex="-1"></a>           x,</span>
<span id="cb7-65"><a href="#cb7-65" aria-hidden="true" tabindex="-1"></a>           <span class="at">id =</span> id,</span>
<span id="cb7-66"><a href="#cb7-66" aria-hidden="true" tabindex="-1"></a>           <span class="at">split =</span> y,</span>
<span id="cb7-67"><a href="#cb7-67" aria-hidden="true" tabindex="-1"></a>           <span class="at">value =</span> n</span>
<span id="cb7-68"><a href="#cb7-68" aria-hidden="true" tabindex="-1"></a>         )) <span class="sc">+</span></span>
<span id="cb7-69"><a href="#cb7-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_parallel_sets</span>(<span class="fu">aes</span>(<span class="at">fill =</span> <span class="sc">!!</span><span class="fu">as.name</span>(fill_lane)), <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">axis.width =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb7-70"><a href="#cb7-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_parallel_sets_axes</span>(<span class="at">axis.width =</span> <span class="fl">0.2</span>, </span>
<span id="cb7-71"><a href="#cb7-71" aria-hidden="true" tabindex="-1"></a>                          <span class="at">fill =</span> <span class="st">"seashell"</span>, </span>
<span id="cb7-72"><a href="#cb7-72" aria-hidden="true" tabindex="-1"></a>                          <span class="at">color =</span> <span class="st">"seashell4"</span>) <span class="sc">+</span></span>
<span id="cb7-73"><a href="#cb7-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_parallel_sets_labels</span>(<span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb7-74"><a href="#cb7-74" aria-hidden="true" tabindex="-1"></a>                            <span class="at">size =</span> <span class="dv">3</span>,</span>
<span id="cb7-75"><a href="#cb7-75" aria-hidden="true" tabindex="-1"></a>                            <span class="at">angle =</span> <span class="dv">0</span>) <span class="sc">+</span> </span>
<span id="cb7-76"><a href="#cb7-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb7-77"><a href="#cb7-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb7-78"><a href="#cb7-78" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb7-79"><a href="#cb7-79" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb7-80"><a href="#cb7-80" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.line =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb7-81"><a href="#cb7-81" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>()</span>
<span id="cb7-82"><a href="#cb7-82" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb7-83"><a href="#cb7-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb7-84"><a href="#cb7-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb7-85"><a href="#cb7-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb7-86"><a href="#cb7-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(</span>
<span id="cb7-87"><a href="#cb7-87" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"segment"</span>,</span>
<span id="cb7-88"><a href="#cb7-88" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> (<span class="fl">3.25</span> <span class="sc">-</span> adjust.scale.pos),</span>
<span id="cb7-89"><a href="#cb7-89" aria-hidden="true" tabindex="-1"></a>    <span class="at">xend =</span> (<span class="fl">3.25</span> <span class="sc">-</span> adjust.scale.pos),</span>
<span id="cb7-90"><a href="#cb7-90" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="dv">20</span>,</span>
<span id="cb7-91"><a href="#cb7-91" aria-hidden="true" tabindex="-1"></a>    <span class="at">yend =</span> <span class="dv">120</span>,</span>
<span id="cb7-92"><a href="#cb7-92" aria-hidden="true" tabindex="-1"></a>    <span class="at">lwd =</span> <span class="dv">2</span></span>
<span id="cb7-93"><a href="#cb7-93" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb7-94"><a href="#cb7-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(</span>
<span id="cb7-95"><a href="#cb7-95" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"text"</span>,</span>
<span id="cb7-96"><a href="#cb7-96" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> (<span class="fl">3.19</span> <span class="sc">-</span> adjust.scale.pos),</span>
<span id="cb7-97"><a href="#cb7-97" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="dv">70</span>,</span>
<span id="cb7-98"><a href="#cb7-98" aria-hidden="true" tabindex="-1"></a>    <span class="at">angle =</span> <span class="dv">90</span>,</span>
<span id="cb7-99"><a href="#cb7-99" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">5</span>,</span>
<span id="cb7-100"><a href="#cb7-100" aria-hidden="true" tabindex="-1"></a>    <span class="at">hjust =</span> <span class="fl">0.5</span>,</span>
<span id="cb7-101"><a href="#cb7-101" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"100 AOIs"</span></span>
<span id="cb7-102"><a href="#cb7-102" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb7-103"><a href="#cb7-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-104"><a href="#cb7-104" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(sankey.plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-sankey" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/fig-sankey-1.png" class="img-fluid figure-img" width="1152"></p>
<figcaption class="figure-caption">Figure&nbsp;1: Sankey Plot</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="#fig-aoibarplot">Figure&nbsp;2</a> shows the total AOI counts per annotation</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>AOI.counts <span class="ot">&lt;-</span> sankey.count.data</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>AOI.counts<span class="sc">$</span>AOI_count <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(AOI.counts<span class="sc">$</span>n)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>AOI.counts<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="fu">as.character</span>(AOI.counts<span class="sc">$</span>x)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>AOI.counts<span class="sc">$</span>annotation <span class="ot">&lt;-</span> AOI.counts<span class="sc">$</span>y</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>AOI.annotation.sum <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">0</span>))</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(AOI.annotation.sum) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"annotation"</span>, <span class="st">"AOI_sum"</span>)</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a data frame of AOI sums per annotation </span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(anno <span class="cf">in</span> <span class="fu">unique</span>(AOI.counts<span class="sc">$</span>annotation)){</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter for a specific annotation</span></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>  anno.subset <span class="ot">&lt;-</span> AOI.counts <span class="sc">%&gt;%</span> </span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(annotation <span class="sc">==</span> anno)</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add together the AOI counts</span></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>  anno.sum.row <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">AOI_sum =</span> <span class="fu">sum</span>(anno.subset<span class="sc">$</span>AOI_count), <span class="at">annotation =</span> anno)</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Append to the master AOI sum df</span></span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>  AOI.annotation.sum <span class="ot">&lt;-</span> <span class="fu">rbind</span>(AOI.annotation.sum, anno.sum.row)</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a>AOI.counts.all <span class="ot">&lt;-</span> <span class="fu">merge</span>(AOI.annotation.sum, AOI.counts, <span class="at">by =</span> <span class="st">"annotation"</span>)</span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a>AOI.counts.all  <span class="ot">&lt;-</span>  AOI.counts.all <span class="sc">%&gt;%</span> </span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">"AOI_sum"</span>, <span class="st">"type"</span>, <span class="st">"annotation"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a>AOI.count.plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(AOI.counts.all, <span class="fu">aes</span>(<span class="at">x =</span> annotation, <span class="at">y =</span> AOI_sum)) <span class="sc">+</span> </span>
<span id="cb8-32"><a href="#cb8-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span> </span>
<span id="cb8-33"><a href="#cb8-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> type, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">scales =</span> <span class="st">"free_x"</span>) <span class="sc">+</span> </span>
<span id="cb8-34"><a href="#cb8-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">30</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb8-35"><a href="#cb8-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> AOI_sum), <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.3</span>, <span class="at">size =</span> <span class="fl">3.5</span>) <span class="sc">+</span></span>
<span id="cb8-36"><a href="#cb8-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, <span class="at">y =</span> <span class="st">"AOI Count"</span>) <span class="sc">+</span> </span>
<span id="cb8-37"><a href="#cb8-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="fu">max</span>(AOI.counts.all<span class="sc">$</span>AOI_sum) <span class="sc">+</span> <span class="dv">30</span>)</span>
<span id="cb8-38"><a href="#cb8-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-39"><a href="#cb8-39" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(AOI.count.plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-aoibarplot" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/fig-aoibarplot-1.png" class="img-fluid figure-img" width="1152"></p>
<figcaption class="figure-caption">Figure&nbsp;2: AOI Count Bar Plot</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="#fig-upsetr">Figure&nbsp;3</a> shows the size of annotation groups</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Vector of all values for upsetr plot</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>all.lane.values <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Gather all of the values for the upsetr plot</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(lane <span class="cf">in</span> lanes){ </span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>  lane.values <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">pData</span>(init.object)[[lane]])</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>  all.lane.values <span class="ot">&lt;-</span> <span class="fu">c</span>(all.lane.values, lane.values)</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the upset df with all FALSE values</span></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>upset.df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">matrix</span>(<span class="cn">FALSE</span>, <span class="at">nrow =</span> <span class="fu">nrow</span>(<span class="fu">pData</span>(init.object)), <span class="at">ncol =</span> <span class="fu">length</span>(all.lane.values)))</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename the columns to be all possible values for the upsetr plot</span></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(upset.df) <span class="ot">&lt;-</span> all.lane.values</span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset the annotation for only the relevant columns for upsetr</span></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a>anno.subset <span class="ot">&lt;-</span> <span class="fu">pData</span>(init.object) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">all_of</span>(lanes))</span>
<span id="cb9-21"><a href="#cb9-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-22"><a href="#cb9-22" aria-hidden="true" tabindex="-1"></a><span class="co"># For each row in the annotation data, if it contains the value of a column in the upsetr plot mark as TRUE</span></span>
<span id="cb9-23"><a href="#cb9-23" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(anno.subset)) {</span>
<span id="cb9-24"><a href="#cb9-24" aria-hidden="true" tabindex="-1"></a>  row_values <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">unlist</span>(anno.subset[i, ]))</span>
<span id="cb9-25"><a href="#cb9-25" aria-hidden="true" tabindex="-1"></a>  upset.df[i, row_values] <span class="ot">&lt;-</span> <span class="cn">TRUE</span></span>
<span id="cb9-26"><a href="#cb9-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb9-27"><a href="#cb9-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-28"><a href="#cb9-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the UpSetR Plot</span></span>
<span id="cb9-29"><a href="#cb9-29" aria-hidden="true" tabindex="-1"></a>AOI.inter.count.plot <span class="ot">&lt;-</span> <span class="fu">upset</span>(upset.df,  </span>
<span id="cb9-30"><a href="#cb9-30" aria-hidden="true" tabindex="-1"></a>                              <span class="at">intersect =</span> all.lane.values, </span>
<span id="cb9-31"><a href="#cb9-31" aria-hidden="true" tabindex="-1"></a>                              <span class="at">width_ratio =</span> <span class="fl">0.4</span>, </span>
<span id="cb9-32"><a href="#cb9-32" aria-hidden="true" tabindex="-1"></a>                              <span class="at">min_size =</span> <span class="dv">4</span>, </span>
<span id="cb9-33"><a href="#cb9-33" aria-hidden="true" tabindex="-1"></a>                              <span class="at">set_sizes=</span>(<span class="fu">upset_set_size</span>() <span class="sc">+</span> </span>
<span id="cb9-34"><a href="#cb9-34" aria-hidden="true" tabindex="-1"></a>                                           <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label=</span>..count..),</span>
<span id="cb9-35"><a href="#cb9-35" aria-hidden="true" tabindex="-1"></a>                                                     <span class="at">hjust=</span><span class="fl">1.1</span>, <span class="at">stat=</span><span class="st">'count'</span>) <span class="sc">+</span></span>
<span id="cb9-36"><a href="#cb9-36" aria-hidden="true" tabindex="-1"></a>                                           <span class="fu">expand_limits</span>(<span class="at">y=</span><span class="fu">nrow</span>(upset.df)) <span class="sc">+</span></span>
<span id="cb9-37"><a href="#cb9-37" aria-hidden="true" tabindex="-1"></a>                                           <span class="fu">theme</span>(<span class="at">axis.text.x=</span><span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">90</span>))))</span>
<span id="cb9-38"><a href="#cb9-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-39"><a href="#cb9-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-40"><a href="#cb9-40" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(AOI.inter.count.plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-upsetr" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/fig-upsetr-1.png" class="img-fluid figure-img" width="1152"></p>
<figcaption class="figure-caption">Figure&nbsp;3: UpSetR Plot</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="qc-and-filtering" class="level2">
<h2 class="anchored" data-anchor-id="qc-and-filtering">QC and Filtering</h2>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>qc.output <span class="ot">&lt;-</span>  <span class="fu">qcProc</span>(<span class="at">object =</span> init.object,</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">min.segment.reads =</span> <span class="dv">1000</span>, </span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">percent.trimmed =</span> <span class="dv">80</span>,    </span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">percent.stitched =</span> <span class="dv">80</span>,   </span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">percent.aligned =</span> <span class="dv">80</span>,    </span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">percent.saturation =</span> <span class="dv">50</span>, </span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">min.negative.count =</span> <span class="dv">3</span>,   </span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">max.ntc.count =</span> <span class="dv">1000</span>,     </span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>                        <span class="at">min.nuclei =</span> <span class="dv">200</span>,         </span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>                        <span class="at">min.area =</span> <span class="dv">1000</span>,</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>                        <span class="at">print.plots =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in qcProc(object = init.object, min.segment.reads = 1000, percent.trimmed = 80, : nuclei, area not found in the annotation, min.nuclei, min.area will not be considered</code></pre>
</div>
<div class="cell-output-display">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/QC%20and%20Filtering-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>

Table: Summary for the NTC values

|NTC Count | # of Segments|
|:---------|-------------:|
|0         |           171|


Table: QC Summary for each Segment

|              | Pass| Warning|
|:-------------|----:|-------:|
|LowReads      |  171|       0|
|LowTrimmed    |  171|       0|
|LowStitched   |  171|       0|
|LowAligned    |  170|       1|
|LowSaturation |  170|       1|
|LowNegatives  |  171|       0|
|HighNTC       |  171|       0|
|TOTAL FLAGS   |  169|       2|


Table: Summary for Segment QC Removal

|         | # Before Removal| # After Removal|
|:--------|----------------:|---------------:|
|Features |            18815|           18815|
|Samples  |              171|             169|


Table: Summary for Probe QC Calls (Grubb's Outlier Test)

| Passed| Global| Local|
|------:|------:|-----:|
|  18807|      1|     7|


Table: Summary for Probe QC Removal

|         | # Before Collapsing| # After Collapsing|
|:--------|-------------------:|------------------:|
|Features |               18815|              18814|
|Samples  |                 169|                169|


Table: Summary for Gene-level Counts

|         | # Before Collapsing| # After Collapsing|
|:--------|-------------------:|------------------:|
|Features |               18814|              18677|
|Samples  |                 169|                169|</code></pre>
</div>
</div>
<p>Summary of QC for AOIs and Probes</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>qc.output<span class="sc">$</span>table</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>$NTC_Count
  NTC_Count # Segments
1         0        171

$QC_Segment
      Parameter Pass Warning
1      LowReads  171       0
2    LowTrimmed  171       0
3   LowStitched  171       0
4    LowAligned  170       1
5 LowSaturation  170       1
6  LowNegatives  171       0
7       HighNTC  171       0
8   TOTAL FLAGS  169       2

$Segment_QC_Removal
   element size.before size.after
1 Features       18815      18815
2  Samples         171        169

$Probe_QC_Calls
  Passed Global Local
1  18807      1     7

$Probe_QC_Removals
         # Probes Before Collapsing # Probes After Collapsing
Features                      18815                     18814
Samples                         169                       169

$Gene_level_Counts
         # Genes Before Collapsing # Genes After Collapsing
Features                     18814                    18677
Samples                        169                      169</code></pre>
</div>
</div>
<section id="aoi-qc" class="level4">
<h4 class="anchored" data-anchor-id="aoi-qc">AOI QC</h4>
<p>AOI distribution by parameter and annotation</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Print AOI plots</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>qc.output<span class="sc">$</span>plot<span class="sc">$</span>trimmed</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/AOI%20Plots-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>qc.output<span class="sc">$</span>plot<span class="sc">$</span>aligned</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/AOI%20Plots-2.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>qc.output<span class="sc">$</span>plot<span class="sc">$</span>stitched</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/AOI%20Plots-3.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>qc.output<span class="sc">$</span>plot<span class="sc">$</span>saturated</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/AOI%20Plots-4.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>qc.output<span class="sc">$</span>plot<span class="sc">$</span>neg.plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/AOI%20Plots-5.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>AOIs that have been flagged with the given QC parameters</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Print AOI flags</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>flag.column.detect <span class="ot">&lt;-</span> <span class="fu">sapply</span>(qc.output<span class="sc">$</span>segment.flags, is.logical)</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>flag.column.names <span class="ot">&lt;-</span> <span class="fu">names</span>(qc.output<span class="sc">$</span>segment.flags[flag.column.detect])</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="co"># A function for coloring TRUE flags as red</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>red.flag <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> <span class="fu">as.logical</span>(x)</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ifelse</span>(x, <span class="st">"red"</span>, <span class="st">"white"</span>) </span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the table using the flag coloring function</span></span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>qc.output<span class="sc">$</span>segment.flags <span class="sc">%&gt;%</span> </span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gt</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data_color</span>(<span class="at">columns =</span> flag.column.names, </span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>                 <span class="at">fn =</span> red.flag, </span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>                 <span class="at">alpha =</span> <span class="fl">0.7</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div id="npfsvchwxa" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#npfsvchwxa table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#npfsvchwxa thead, #npfsvchwxa tbody, #npfsvchwxa tfoot, #npfsvchwxa tr, #npfsvchwxa td, #npfsvchwxa th {
  border-style: none;
}

#npfsvchwxa p {
  margin: 0;
  padding: 0;
}

#npfsvchwxa .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#npfsvchwxa .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#npfsvchwxa .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#npfsvchwxa .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#npfsvchwxa .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#npfsvchwxa .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#npfsvchwxa .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#npfsvchwxa .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#npfsvchwxa .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#npfsvchwxa .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#npfsvchwxa .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#npfsvchwxa .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#npfsvchwxa .gt_spanner_row {
  border-bottom-style: hidden;
}

#npfsvchwxa .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#npfsvchwxa .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#npfsvchwxa .gt_from_md > :first-child {
  margin-top: 0;
}

#npfsvchwxa .gt_from_md > :last-child {
  margin-bottom: 0;
}

#npfsvchwxa .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#npfsvchwxa .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#npfsvchwxa .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#npfsvchwxa .gt_row_group_first td {
  border-top-width: 2px;
}

#npfsvchwxa .gt_row_group_first th {
  border-top-width: 2px;
}

#npfsvchwxa .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#npfsvchwxa .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#npfsvchwxa .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#npfsvchwxa .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#npfsvchwxa .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#npfsvchwxa .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#npfsvchwxa .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#npfsvchwxa .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#npfsvchwxa .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#npfsvchwxa .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#npfsvchwxa .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#npfsvchwxa .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#npfsvchwxa .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#npfsvchwxa .gt_left {
  text-align: left;
}

#npfsvchwxa .gt_center {
  text-align: center;
}

#npfsvchwxa .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#npfsvchwxa .gt_font_normal {
  font-weight: normal;
}

#npfsvchwxa .gt_font_bold {
  font-weight: bold;
}

#npfsvchwxa .gt_font_italic {
  font-style: italic;
}

#npfsvchwxa .gt_super {
  font-size: 65%;
}

#npfsvchwxa .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#npfsvchwxa .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#npfsvchwxa .gt_indent_1 {
  text-indent: 5px;
}

#npfsvchwxa .gt_indent_2 {
  text-indent: 10px;
}

#npfsvchwxa .gt_indent_3 {
  text-indent: 15px;
}

#npfsvchwxa .gt_indent_4 {
  text-indent: 20px;
}

#npfsvchwxa .gt_indent_5 {
  text-indent: 25px;
}

#npfsvchwxa .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#npfsvchwxa div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="SampleID">SampleID</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="segmentID">segmentID</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="RawReadCount">RawReadCount</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="LowReads">LowReads</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="TrimmedPerc">TrimmedPerc</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="LowTrimmed">LowTrimmed</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="StitchedPerc">StitchedPerc</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="LowStitched">LowStitched</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="AlignedPerc">AlignedPerc</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="LowAligned">LowAligned</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="SaturatedPerc">SaturatedPerc</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="LowSaturation">LowSaturation</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="NegGeoMean">NegGeoMean</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id="LowNegatives">LowNegatives</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="SampleID" class="gt_row gt_left">DSP-1001660014796-H-D03.dcc</td>
<td headers="segmentID" class="gt_row gt_left">Biopsy_G|vessel|full_ROI|slide_4|9</td>
<td headers="RawReadCount" class="gt_row gt_right">2537192</td>
<td headers="LowReads" class="gt_row gt_center" style="background-color: rgba(255,255,255,0.7); color: #000000;">FALSE</td>
<td headers="TrimmedPerc" class="gt_row gt_right">99.59727</td>
<td headers="LowTrimmed" class="gt_row gt_center" style="background-color: rgba(255,255,255,0.7); color: #000000;">FALSE</td>
<td headers="StitchedPerc" class="gt_row gt_right">98.97820</td>
<td headers="LowStitched" class="gt_row gt_center" style="background-color: rgba(255,255,255,0.7); color: #000000;">FALSE</td>
<td headers="AlignedPerc" class="gt_row gt_right">67.80157</td>
<td headers="LowAligned" class="gt_row gt_center" style="background-color: rgba(255,0,0,0.7); color: #FFFFFF;">TRUE</td>
<td headers="SaturatedPerc" class="gt_row gt_right">85.57697</td>
<td headers="LowSaturation" class="gt_row gt_center" style="background-color: rgba(255,255,255,0.7); color: #000000;">FALSE</td>
<td headers="NegGeoMean" class="gt_row gt_right">7.458466</td>
<td headers="LowNegatives" class="gt_row gt_center" style="background-color: rgba(255,255,255,0.7); color: #000000;">FALSE</td></tr>
    <tr><td headers="SampleID" class="gt_row gt_left">DSP-1001660014797-G-A09.dcc</td>
<td headers="segmentID" class="gt_row gt_left">Biopsy_S|tumor|LANA_pos|slide_1B|8</td>
<td headers="RawReadCount" class="gt_row gt_right">401063</td>
<td headers="LowReads" class="gt_row gt_center" style="background-color: rgba(255,255,255,0.7); color: #000000;">FALSE</td>
<td headers="TrimmedPerc" class="gt_row gt_right">99.59956</td>
<td headers="LowTrimmed" class="gt_row gt_center" style="background-color: rgba(255,255,255,0.7); color: #000000;">FALSE</td>
<td headers="StitchedPerc" class="gt_row gt_right">98.81964</td>
<td headers="LowStitched" class="gt_row gt_center" style="background-color: rgba(255,255,255,0.7); color: #000000;">FALSE</td>
<td headers="AlignedPerc" class="gt_row gt_right">96.53072</td>
<td headers="LowAligned" class="gt_row gt_center" style="background-color: rgba(255,255,255,0.7); color: #000000;">FALSE</td>
<td headers="SaturatedPerc" class="gt_row gt_right">45.46027</td>
<td headers="LowSaturation" class="gt_row gt_center" style="background-color: rgba(255,0,0,0.7); color: #FFFFFF;">TRUE</td>
<td headers="NegGeoMean" class="gt_row gt_right">5.003674</td>
<td headers="LowNegatives" class="gt_row gt_center" style="background-color: rgba(255,255,255,0.7); color: #000000;">FALSE</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</div>
</section>
<section id="probe-qc" class="level3">
<h3 class="anchored" data-anchor-id="probe-qc">Probe QC</h3>
<p>Probes that have been flagged as either local or global outliers.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the table for probe flags</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>probe.flags.df <span class="ot">&lt;-</span> qc.output<span class="sc">$</span>probe.flags <span class="sc">%&gt;%</span> <span class="fu">separate_rows</span>(LocalFlag, <span class="at">sep =</span> <span class="st">","</span>)</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename the dcc file name column</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>probe.flags.df<span class="sc">$</span>Sample_ID <span class="ot">&lt;-</span> probe.flags.df<span class="sc">$</span>LocalFlag</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Grab the annotation for only the columns to map</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>annotation <span class="ot">&lt;-</span> <span class="fu">pData</span>(qc.output<span class="sc">$</span>object)</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>annotation<span class="sc">$</span>Sample_ID <span class="ot">&lt;-</span> <span class="fu">rownames</span>(annotation)</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>annotation.subset <span class="ot">&lt;-</span> annotation <span class="sc">%&gt;%</span> </span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Sample_ID, segmentID)</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Map the AOI names in the flags to the segmentID</span></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>probe.flags.df <span class="ot">&lt;-</span> <span class="fu">merge</span>(probe.flags.df, annotation.subset, <span class="at">by =</span> <span class="st">"Sample_ID"</span>)</span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove the dcc file name column </span></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>probe.flags.table <span class="ot">&lt;-</span> probe.flags.df <span class="sc">%&gt;%</span> </span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(TargetName, RTS_ID, segmentID, FlagType)</span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a><span class="co"># For a summary of only probe names</span></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>probe.flag.summary <span class="ot">&lt;-</span> qc.output<span class="sc">$</span>probe.flags <span class="sc">%&gt;%</span> </span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(TargetName, RTS_ID, FlagType)</span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Toggle to include all flags or only summary</span></span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a>include.all <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb21-28"><a href="#cb21-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-29"><a href="#cb21-29" aria-hidden="true" tabindex="-1"></a><span class="co"># For all flags including segment ID name</span></span>
<span id="cb21-30"><a href="#cb21-30" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(include.all <span class="sc">==</span> <span class="cn">TRUE</span>){ </span>
<span id="cb21-31"><a href="#cb21-31" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb21-32"><a href="#cb21-32" aria-hidden="true" tabindex="-1"></a>  probe.flags.table <span class="sc">%&gt;%</span> </span>
<span id="cb21-33"><a href="#cb21-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gt</span>()</span>
<span id="cb21-34"><a href="#cb21-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb21-35"><a href="#cb21-35" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb21-36"><a href="#cb21-36" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb21-37"><a href="#cb21-37" aria-hidden="true" tabindex="-1"></a>  probe.flag.summary <span class="sc">%&gt;%</span> </span>
<span id="cb21-38"><a href="#cb21-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gt</span>()</span>
<span id="cb21-39"><a href="#cb21-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb21-40"><a href="#cb21-40" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div id="grxytcspck" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#grxytcspck table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#grxytcspck thead, #grxytcspck tbody, #grxytcspck tfoot, #grxytcspck tr, #grxytcspck td, #grxytcspck th {
  border-style: none;
}

#grxytcspck p {
  margin: 0;
  padding: 0;
}

#grxytcspck .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#grxytcspck .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#grxytcspck .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#grxytcspck .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#grxytcspck .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#grxytcspck .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#grxytcspck .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#grxytcspck .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#grxytcspck .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#grxytcspck .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#grxytcspck .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#grxytcspck .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#grxytcspck .gt_spanner_row {
  border-bottom-style: hidden;
}

#grxytcspck .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#grxytcspck .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#grxytcspck .gt_from_md > :first-child {
  margin-top: 0;
}

#grxytcspck .gt_from_md > :last-child {
  margin-bottom: 0;
}

#grxytcspck .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#grxytcspck .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#grxytcspck .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#grxytcspck .gt_row_group_first td {
  border-top-width: 2px;
}

#grxytcspck .gt_row_group_first th {
  border-top-width: 2px;
}

#grxytcspck .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#grxytcspck .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#grxytcspck .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#grxytcspck .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#grxytcspck .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#grxytcspck .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#grxytcspck .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#grxytcspck .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#grxytcspck .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#grxytcspck .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#grxytcspck .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#grxytcspck .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#grxytcspck .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#grxytcspck .gt_left {
  text-align: left;
}

#grxytcspck .gt_center {
  text-align: center;
}

#grxytcspck .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#grxytcspck .gt_font_normal {
  font-weight: normal;
}

#grxytcspck .gt_font_bold {
  font-weight: bold;
}

#grxytcspck .gt_font_italic {
  font-style: italic;
}

#grxytcspck .gt_super {
  font-size: 65%;
}

#grxytcspck .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#grxytcspck .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#grxytcspck .gt_indent_1 {
  text-indent: 5px;
}

#grxytcspck .gt_indent_2 {
  text-indent: 10px;
}

#grxytcspck .gt_indent_3 {
  text-indent: 15px;
}

#grxytcspck .gt_indent_4 {
  text-indent: 20px;
}

#grxytcspck .gt_indent_5 {
  text-indent: 25px;
}

#grxytcspck .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#grxytcspck div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="TargetName">TargetName</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="RTS_ID">RTS_ID</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="FlagType">FlagType</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="TargetName" class="gt_row gt_left">NegProbe-WTX</td>
<td headers="RTS_ID" class="gt_row gt_left">RTS0039321</td>
<td headers="FlagType" class="gt_row gt_left">Local</td></tr>
    <tr><td headers="TargetName" class="gt_row gt_left">NegProbe-WTX</td>
<td headers="RTS_ID" class="gt_row gt_left">RTS0039332</td>
<td headers="FlagType" class="gt_row gt_left">Local</td></tr>
    <tr><td headers="TargetName" class="gt_row gt_left">NegProbe-WTX</td>
<td headers="RTS_ID" class="gt_row gt_left">RTS0039334</td>
<td headers="FlagType" class="gt_row gt_left">Local</td></tr>
    <tr><td headers="TargetName" class="gt_row gt_left">NegProbe-WTX</td>
<td headers="RTS_ID" class="gt_row gt_left">RTS0039339</td>
<td headers="FlagType" class="gt_row gt_left">Local</td></tr>
    <tr><td headers="TargetName" class="gt_row gt_left">NegProbe-WTX</td>
<td headers="RTS_ID" class="gt_row gt_left">RTS0039346</td>
<td headers="FlagType" class="gt_row gt_left">Global</td></tr>
    <tr><td headers="TargetName" class="gt_row gt_left">NegProbe-WTX</td>
<td headers="RTS_ID" class="gt_row gt_left">RTS0039349</td>
<td headers="FlagType" class="gt_row gt_left">Local</td></tr>
    <tr><td headers="TargetName" class="gt_row gt_left">NegProbe-WTX</td>
<td headers="RTS_ID" class="gt_row gt_left">RTS0039367</td>
<td headers="FlagType" class="gt_row gt_left">Local</td></tr>
    <tr><td headers="TargetName" class="gt_row gt_left">NegProbe-WTX</td>
<td headers="RTS_ID" class="gt_row gt_left">RTS0051994</td>
<td headers="FlagType" class="gt_row gt_left">Local</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</div>
</section>
<section id="filtering" class="level3">
<h3 class="anchored" data-anchor-id="filtering">Filtering</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>object <span class="ot">&lt;-</span> qc.output<span class="sc">$</span>object</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up lists of segment IDs</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>segment.list.total <span class="ot">&lt;-</span> <span class="fu">pData</span>(object)<span class="sc">$</span>segmentID</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Define Modules</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>modules <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".pkc"</span>, <span class="st">""</span>, pkc.file.name)</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate limit of quantification (LOQ) in each segment</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="co"># LOQ = geomean(NegProbes) * geoSD(NegProbes)^(LOQ cutoff)</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a><span class="co"># LOQ is calculated for each module (pkc file)</span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>loq <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">row.names =</span> <span class="fu">colnames</span>(object))</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>loq.min <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>loq.cutoff <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(module <span class="cf">in</span> modules) {</span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>  vars <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">c</span>(<span class="st">"NegGeoMean_"</span>, <span class="st">"NegGeoSD_"</span>),</span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>                 module)</span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">all</span>(vars[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>] <span class="sc">%in%</span> <span class="fu">colnames</span>(<span class="fu">pData</span>(object)))) {</span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a>    neg.geo.mean <span class="ot">&lt;-</span> vars[<span class="dv">1</span>]</span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a>    neg.geo.sd <span class="ot">&lt;-</span> vars[<span class="dv">2</span>]</span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a>    loq[, module] <span class="ot">&lt;-</span></span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a>      <span class="fu">pmax</span>(loq.min,</span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a>           <span class="fu">pData</span>(object)[, neg.geo.mean] <span class="sc">*</span> </span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a>             <span class="fu">pData</span>(object)[, neg.geo.sd] <span class="sc">^</span> loq.cutoff)</span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-32"><a href="#cb22-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Store the loq df in the annotation df</span></span>
<span id="cb22-33"><a href="#cb22-33" aria-hidden="true" tabindex="-1"></a><span class="fu">pData</span>(object)<span class="sc">$</span>loq <span class="ot">&lt;-</span> loq</span>
<span id="cb22-34"><a href="#cb22-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-35"><a href="#cb22-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Setup a master loq matrix</span></span>
<span id="cb22-36"><a href="#cb22-36" aria-hidden="true" tabindex="-1"></a>loq.mat <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb22-37"><a href="#cb22-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-38"><a href="#cb22-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-39"><a href="#cb22-39" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(module <span class="cf">in</span> modules) {</span>
<span id="cb22-40"><a href="#cb22-40" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Gather rows with the given module</span></span>
<span id="cb22-41"><a href="#cb22-41" aria-hidden="true" tabindex="-1"></a>  ind <span class="ot">&lt;-</span> <span class="fu">fData</span>(object)<span class="sc">$</span>Module <span class="sc">==</span> module</span>
<span id="cb22-42"><a href="#cb22-42" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-43"><a href="#cb22-43" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if each feature has counts above the LOQ</span></span>
<span id="cb22-44"><a href="#cb22-44" aria-hidden="true" tabindex="-1"></a>  mat.i <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">esApply</span>(object[ind, ], <span class="at">MARGIN =</span> <span class="dv">1</span>,</span>
<span id="cb22-45"><a href="#cb22-45" aria-hidden="true" tabindex="-1"></a>                     <span class="at">FUN =</span> <span class="cf">function</span>(x) {</span>
<span id="cb22-46"><a href="#cb22-46" aria-hidden="true" tabindex="-1"></a>                       x <span class="sc">&gt;</span> loq[, module]</span>
<span id="cb22-47"><a href="#cb22-47" aria-hidden="true" tabindex="-1"></a>                     }))</span>
<span id="cb22-48"><a href="#cb22-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-49"><a href="#cb22-49" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Store results in the master loq matrix</span></span>
<span id="cb22-50"><a href="#cb22-50" aria-hidden="true" tabindex="-1"></a>  loq.mat <span class="ot">&lt;-</span> <span class="fu">rbind</span>(loq.mat, mat.i)</span>
<span id="cb22-51"><a href="#cb22-51" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb22-52"><a href="#cb22-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-53"><a href="#cb22-53" aria-hidden="true" tabindex="-1"></a><span class="co"># ensure ordering since this is stored outside of the geomxSet</span></span>
<span id="cb22-54"><a href="#cb22-54" aria-hidden="true" tabindex="-1"></a>loq.mat <span class="ot">&lt;-</span> loq.mat[<span class="fu">fData</span>(object)<span class="sc">$</span>TargetName, ]</span>
<span id="cb22-55"><a href="#cb22-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-56"><a href="#cb22-56" aria-hidden="true" tabindex="-1"></a><span class="co"># Evaluate and Filter Segment Gene Detection Rate</span></span>
<span id="cb22-57"><a href="#cb22-57" aria-hidden="true" tabindex="-1"></a><span class="co"># Save detection rate information to pheno data</span></span>
<span id="cb22-58"><a href="#cb22-58" aria-hidden="true" tabindex="-1"></a><span class="fu">pData</span>(object)<span class="sc">$</span>GenesDetected <span class="ot">&lt;-</span> <span class="fu">colSums</span>(loq.mat, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb22-59"><a href="#cb22-59" aria-hidden="true" tabindex="-1"></a><span class="fu">pData</span>(object)<span class="sc">$</span>GeneDetectionRate <span class="ot">&lt;-</span> <span class="dv">100</span><span class="sc">*</span>(<span class="fu">pData</span>(object)<span class="sc">$</span>GenesDetected <span class="sc">/</span> <span class="fu">nrow</span>(object))</span>
<span id="cb22-60"><a href="#cb22-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-61"><a href="#cb22-61" aria-hidden="true" tabindex="-1"></a><span class="co"># Establish detection bins</span></span>
<span id="cb22-62"><a href="#cb22-62" aria-hidden="true" tabindex="-1"></a>detection.bins <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"&lt;1"</span>, <span class="st">"1-5"</span>, <span class="st">"5-10"</span>, <span class="st">"10-15"</span>, <span class="st">"&gt;15"</span>)</span>
<span id="cb22-63"><a href="#cb22-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-64"><a href="#cb22-64" aria-hidden="true" tabindex="-1"></a><span class="co"># Determine detection thresholds: 1%, 5%, 10%, 15%, &gt;15%</span></span>
<span id="cb22-65"><a href="#cb22-65" aria-hidden="true" tabindex="-1"></a><span class="fu">pData</span>(object)<span class="sc">$</span>DetectionThreshold <span class="ot">&lt;-</span> </span>
<span id="cb22-66"><a href="#cb22-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cut</span>(<span class="fu">pData</span>(object)<span class="sc">$</span>GeneDetectionRate,</span>
<span id="cb22-67"><a href="#cb22-67" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">15</span>, <span class="dv">100</span>),</span>
<span id="cb22-68"><a href="#cb22-68" aria-hidden="true" tabindex="-1"></a>      <span class="at">labels =</span> detection.bins)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="overall-gene-detection-per-aoi" class="level4">
<h4 class="anchored" data-anchor-id="overall-gene-detection-per-aoi">Overall Gene Detection per AOI</h4>
<p><a href="#fig-GeneDetectionbByAOI">Figure&nbsp;4</a> shows detection rate per AOI, colored by region.</p>
<div class="cell" data-tbl-cap="Overall Gene Detection per AOI">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># stacked bar plot of different cut points (1%, 5%, 10%, 15%)</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>segment.stacked.bar.plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">pData</span>(object),</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">aes</span>(<span class="at">x =</span> DetectionThreshold)) <span class="sc">+</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">fill =</span> region)) <span class="sc">+</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">stat =</span> <span class="st">"count"</span>, <span class="fu">aes</span>(<span class="at">label =</span> ..count..), <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.1</span>))) <span class="sc">+</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Gene Detection Rate"</span>,</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Segments, #"</span>,</span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">fill =</span> <span class="st">"AOI Annotation"</span>)</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(segment.stacked.bar.plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-GeneDetectionbByAOI" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/fig-GeneDetectionbByAOI-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;4: <strong>?(caption)</strong></figcaption>
</figure>
</div>
</div>
</div>
<p>AOIs in the low detection bin of 1-5%</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># cut percent genes detected at 1, 5, 10, 15</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>segment.table <span class="ot">&lt;-</span> <span class="fu">kable</span>(<span class="fu">table</span>(<span class="fu">pData</span>(object)<span class="sc">$</span>DetectionThreshold, </span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">pData</span>(object)<span class="sc">$</span>class))</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Make a list of segments with low detection</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>low.detection.segments <span class="ot">&lt;-</span> <span class="fu">pData</span>(object) <span class="sc">%&gt;%</span> </span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(GeneDetectionRate <span class="sc">&lt;</span> <span class="dv">5</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">any_of</span>(<span class="fu">c</span>(<span class="st">"segmentID"</span>, <span class="st">"GeneDetectionRate"</span>)))</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(low.detection.segments) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Print low detection segment table</span></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>low.detection.segments <span class="sc">%&gt;%</span> </span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gt</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div id="rhkakfjqnq" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#rhkakfjqnq table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#rhkakfjqnq thead, #rhkakfjqnq tbody, #rhkakfjqnq tfoot, #rhkakfjqnq tr, #rhkakfjqnq td, #rhkakfjqnq th {
  border-style: none;
}

#rhkakfjqnq p {
  margin: 0;
  padding: 0;
}

#rhkakfjqnq .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#rhkakfjqnq .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#rhkakfjqnq .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#rhkakfjqnq .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#rhkakfjqnq .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#rhkakfjqnq .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#rhkakfjqnq .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#rhkakfjqnq .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#rhkakfjqnq .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#rhkakfjqnq .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#rhkakfjqnq .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#rhkakfjqnq .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#rhkakfjqnq .gt_spanner_row {
  border-bottom-style: hidden;
}

#rhkakfjqnq .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#rhkakfjqnq .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#rhkakfjqnq .gt_from_md > :first-child {
  margin-top: 0;
}

#rhkakfjqnq .gt_from_md > :last-child {
  margin-bottom: 0;
}

#rhkakfjqnq .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#rhkakfjqnq .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#rhkakfjqnq .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#rhkakfjqnq .gt_row_group_first td {
  border-top-width: 2px;
}

#rhkakfjqnq .gt_row_group_first th {
  border-top-width: 2px;
}

#rhkakfjqnq .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#rhkakfjqnq .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#rhkakfjqnq .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#rhkakfjqnq .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#rhkakfjqnq .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#rhkakfjqnq .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#rhkakfjqnq .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#rhkakfjqnq .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#rhkakfjqnq .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#rhkakfjqnq .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#rhkakfjqnq .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#rhkakfjqnq .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#rhkakfjqnq .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#rhkakfjqnq .gt_left {
  text-align: left;
}

#rhkakfjqnq .gt_center {
  text-align: center;
}

#rhkakfjqnq .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#rhkakfjqnq .gt_font_normal {
  font-weight: normal;
}

#rhkakfjqnq .gt_font_bold {
  font-weight: bold;
}

#rhkakfjqnq .gt_font_italic {
  font-style: italic;
}

#rhkakfjqnq .gt_super {
  font-size: 65%;
}

#rhkakfjqnq .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#rhkakfjqnq .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#rhkakfjqnq .gt_indent_1 {
  text-indent: 5px;
}

#rhkakfjqnq .gt_indent_2 {
  text-indent: 10px;
}

#rhkakfjqnq .gt_indent_3 {
  text-indent: 15px;
}

#rhkakfjqnq .gt_indent_4 {
  text-indent: 20px;
}

#rhkakfjqnq .gt_indent_5 {
  text-indent: 25px;
}

#rhkakfjqnq .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#rhkakfjqnq div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="segmentID">segmentID</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="GeneDetectionRate">GeneDetectionRate</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="segmentID" class="gt_row gt_left">Biopsy_S|tumor|LANA_pos|slide_3|24</td>
<td headers="GeneDetectionRate" class="gt_row gt_right">4.759865</td></tr>
    <tr><td headers="segmentID" class="gt_row gt_left">Biopsy_S|tumor|LANA_pos|slide_3|25</td>
<td headers="GeneDetectionRate" class="gt_row gt_right">4.749157</td></tr>
    <tr><td headers="segmentID" class="gt_row gt_left">Biopsy_G|vessel|full_ROI|slide_4|10</td>
<td headers="GeneDetectionRate" class="gt_row gt_right">4.020988</td></tr>
    <tr><td headers="segmentID" class="gt_row gt_left">Biopsy_S|vessel|full_ROI|slide_4|44</td>
<td headers="GeneDetectionRate" class="gt_row gt_right">4.797344</td></tr>
    <tr><td headers="segmentID" class="gt_row gt_left">Biopsy_G|vessel|full_ROI|slide_1B|13</td>
<td headers="GeneDetectionRate" class="gt_row gt_right">4.658136</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</div>
<p>Gene detection for all AOIs</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Export a summary of the segment gene detection</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>segment.detection.summary <span class="ot">&lt;-</span> <span class="fu">pData</span>(object) <span class="sc">%&gt;%</span> </span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">any_of</span>(<span class="fu">c</span>(<span class="st">"segmentID"</span>, <span class="st">"GeneDetectionRate"</span>, <span class="st">"DetectionThreshold"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="filter-out-aois-with-low-detection" class="level5">
<h5 class="anchored" data-anchor-id="filter-out-aois-with-low-detection">Filter out AOIs with low detection</h5>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter the data using the cutoff for gene detection rate</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>segment.gene.rate.cutoff <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>object.segment.filtered <span class="ot">&lt;-</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>    object[, <span class="fu">pData</span>(object)<span class="sc">$</span>GeneDetectionRate <span class="sc">&gt;=</span> segment.gene.rate.cutoff]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="detection-per-gene" class="level4">
<h4 class="anchored" data-anchor-id="detection-per-gene">Detection per Gene</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Evaluate and Filter Study-wide Gene Detection Rate </span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate detection rate:</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>loq.mat <span class="ot">&lt;-</span> loq.mat[, <span class="fu">colnames</span>(object.segment.filtered)]</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="fu">fData</span>(object.segment.filtered)<span class="sc">$</span>DetectedSegments <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(loq.mat, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a><span class="fu">fData</span>(object.segment.filtered)<span class="sc">$</span>DetectionRate <span class="ot">&lt;-</span></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>  <span class="dv">100</span><span class="sc">*</span>(<span class="fu">fData</span>(object.segment.filtered)<span class="sc">$</span>DetectedSegments <span class="sc">/</span> <span class="fu">nrow</span>(<span class="fu">pData</span>(object)))</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Establish detection bins</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>detection.bins <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"0"</span>, <span class="st">"&lt;1"</span>, <span class="st">"1-5"</span>, <span class="st">"5-10"</span>, <span class="st">"10-20"</span>, <span class="st">"20-30"</span>, <span class="st">"30-40"</span>, <span class="st">"40-50"</span>, <span class="st">"&gt;50"</span>)</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Determine detection thresholds: 1%, 5%, 10%, 15%, &gt;15%</span></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a><span class="fu">fData</span>(object.segment.filtered)<span class="sc">$</span>DetectionThreshold <span class="ot">&lt;-</span> </span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cut</span>(<span class="fu">fData</span>(object.segment.filtered)<span class="sc">$</span>DetectionRate,</span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">breaks =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">30</span>, <span class="dv">40</span>, <span class="dv">50</span>, <span class="dv">100</span>),</span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>      <span class="at">labels =</span> detection.bins)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><a href="#fig-DetectionPerGene">Figure&nbsp;5</a> shows the percent of all AOIs individual genes are detected within</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>gene.stacked.bar.plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">fData</span>(object.segment.filtered),</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">aes</span>(<span class="at">x =</span> DetectionThreshold)) <span class="sc">+</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">fill =</span> Module)) <span class="sc">+</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">stat =</span> <span class="st">"count"</span>, <span class="fu">aes</span>(<span class="at">label =</span> ..count..), <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.1</span>))) <span class="sc">+</span></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Gene Detection Rate"</span>,</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Genes, #"</span>,</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">fill =</span> <span class="st">"Probe Set"</span>)</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(gene.stacked.bar.plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-DetectionPerGene" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/fig-DetectionPerGene-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;5: Gene Detection Percent of All AOIs</figcaption>
</figure>
</div>
</div>
</div>
<p>Gene detection rates for specified genes of interest</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Gene of interest detection table</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>goi <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"A2M"</span>, <span class="st">"CD44"</span>)</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>goi.table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Gene =</span> goi,</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">Number =</span> <span class="fu">fData</span>(object.segment.filtered)[goi, <span class="st">"DetectedSegments"</span>],</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">DetectionRate =</span> <span class="fu">fData</span>(object.segment.filtered)[goi, <span class="st">"DetectionRate"</span>])</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the GOI table</span></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>goi.table <span class="sc">%&gt;%</span> </span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gt</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="tbl-GenesOfInterest" class="anchored">

<div id="jqcrkfojel" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#jqcrkfojel table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#jqcrkfojel thead, #jqcrkfojel tbody, #jqcrkfojel tfoot, #jqcrkfojel tr, #jqcrkfojel td, #jqcrkfojel th {
  border-style: none;
}

#jqcrkfojel p {
  margin: 0;
  padding: 0;
}

#jqcrkfojel .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#jqcrkfojel .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#jqcrkfojel .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#jqcrkfojel .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#jqcrkfojel .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#jqcrkfojel .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jqcrkfojel .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#jqcrkfojel .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#jqcrkfojel .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#jqcrkfojel .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#jqcrkfojel .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#jqcrkfojel .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#jqcrkfojel .gt_spanner_row {
  border-bottom-style: hidden;
}

#jqcrkfojel .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#jqcrkfojel .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#jqcrkfojel .gt_from_md > :first-child {
  margin-top: 0;
}

#jqcrkfojel .gt_from_md > :last-child {
  margin-bottom: 0;
}

#jqcrkfojel .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#jqcrkfojel .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#jqcrkfojel .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#jqcrkfojel .gt_row_group_first td {
  border-top-width: 2px;
}

#jqcrkfojel .gt_row_group_first th {
  border-top-width: 2px;
}

#jqcrkfojel .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#jqcrkfojel .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#jqcrkfojel .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#jqcrkfojel .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jqcrkfojel .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#jqcrkfojel .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#jqcrkfojel .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#jqcrkfojel .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#jqcrkfojel .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jqcrkfojel .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#jqcrkfojel .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#jqcrkfojel .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#jqcrkfojel .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#jqcrkfojel .gt_left {
  text-align: left;
}

#jqcrkfojel .gt_center {
  text-align: center;
}

#jqcrkfojel .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#jqcrkfojel .gt_font_normal {
  font-weight: normal;
}

#jqcrkfojel .gt_font_bold {
  font-weight: bold;
}

#jqcrkfojel .gt_font_italic {
  font-style: italic;
}

#jqcrkfojel .gt_super {
  font-size: 65%;
}

#jqcrkfojel .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#jqcrkfojel .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#jqcrkfojel .gt_indent_1 {
  text-indent: 5px;
}

#jqcrkfojel .gt_indent_2 {
  text-indent: 10px;
}

#jqcrkfojel .gt_indent_3 {
  text-indent: 15px;
}

#jqcrkfojel .gt_indent_4 {
  text-indent: 20px;
}

#jqcrkfojel .gt_indent_5 {
  text-indent: 25px;
}

#jqcrkfojel .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#jqcrkfojel div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false"><caption>Table&nbsp;1:  <p>Gene of Interest Detection Rate</p> </caption>
  <thead>
    <tr class="gt_col_headings">
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="Gene">Gene</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="Number">Number</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="DetectionRate">DetectionRate</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td headers="Gene" class="gt_row gt_left">A2M</td>
<td headers="Number" class="gt_row gt_right">169</td>
<td headers="DetectionRate" class="gt_row gt_right">100.00000</td></tr>
    <tr><td headers="Gene" class="gt_row gt_left">CD44</td>
<td headers="Number" class="gt_row gt_right">160</td>
<td headers="DetectionRate" class="gt_row gt_right">94.67456</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</div>
</div>
<p><a href="#fig-DetectionPerGeneLoss">Figure&nbsp;6</a> shows the loss of percentage of all AOIs individual genes are detected within</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot detection rate:</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>plot.detect <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Freq =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">30</span>, <span class="dv">50</span>))</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>plot.detect<span class="sc">$</span>Number <span class="ot">&lt;-</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unlist</span>(<span class="fu">lapply</span>(<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">30</span>, <span class="dv">50</span>),</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>                <span class="cf">function</span>(x) {<span class="fu">sum</span>(<span class="fu">fData</span>(object.segment.filtered)<span class="sc">$</span>DetectionRate <span class="sc">&gt;=</span> x)}))</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>plot.detect<span class="sc">$</span>Rate <span class="ot">&lt;-</span> plot.detect<span class="sc">$</span>Number <span class="sc">/</span> <span class="fu">nrow</span>(<span class="fu">fData</span>(object.segment.filtered))</span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(plot.detect) <span class="ot">&lt;-</span> plot.detect<span class="sc">$</span>Freq</span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>genes.detected.plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(plot.detect, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.factor</span>(Freq), <span class="at">y =</span> Rate, <span class="at">fill =</span> Rate)) <span class="sc">+</span></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">formatC</span>(Number, <span class="at">format =</span> <span class="st">"d"</span>, <span class="at">big.mark =</span> <span class="st">","</span>)),</span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>            <span class="at">vjust =</span> <span class="fl">1.6</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient2</span>(<span class="at">low =</span> <span class="st">"orange2"</span>, <span class="at">mid =</span> <span class="st">"lightblue"</span>,</span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a>                       <span class="at">high =</span> <span class="st">"dodgerblue3"</span>, <span class="at">midpoint =</span> <span class="fl">0.65</span>,</span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>                       <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),</span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>                       <span class="at">labels =</span> scales<span class="sc">::</span>percent) <span class="sc">+</span></span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> scales<span class="sc">::</span>percent, <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>),</span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a>                     <span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>))) <span class="sc">+</span></span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"% of Segments"</span>,</span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Genes Detected, % of Panel &gt; loq"</span>)</span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(genes.detected.plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-DetectionPerGeneLoss" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/fig-DetectionPerGeneLoss-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;6: Gene Detection Percent of All AOIs</figcaption>
</figure>
</div>
</div>
</div>
<p>Summary of all gene detection rates</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Gather a summary of the every gene's detection percentage in all AOIs</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>gene.detection.summary <span class="ot">&lt;-</span> <span class="fu">fData</span>(object.segment.filtered) <span class="sc">%&gt;%</span> </span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene =</span> <span class="fu">rownames</span>(<span class="fu">fData</span>(object.segment.filtered))) <span class="sc">%&gt;%</span> </span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">any_of</span>(<span class="fu">c</span>(<span class="st">"gene"</span>, <span class="st">"DetectionRate"</span>, <span class="st">"DetectionThreshold"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="filter-out-genes-with-low-detection" class="level5">
<h5 class="anchored" data-anchor-id="filter-out-genes-with-low-detection">Filter out genes with low detection</h5>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set the cutoff for gene detection</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>study.gene.rate.cutoff <span class="ot">&lt;-</span> <span class="fl">0.00</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset for genes above the study gene detection rate cutoff</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Manually include the negative control probe, for downstream use</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>negative.probe.fData <span class="ot">&lt;-</span> <span class="fu">subset</span>(<span class="fu">fData</span>(object.segment.filtered), CodeClass <span class="sc">==</span> <span class="st">"Negative"</span>)</span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>neg.probes <span class="ot">&lt;-</span> <span class="fu">unique</span>(negative.probe.fData<span class="sc">$</span>TargetName)</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>object.gene.filtered <span class="ot">&lt;-</span> object.segment.filtered[<span class="fu">fData</span>(object.segment.filtered)<span class="sc">$</span>DetectionRate <span class="sc">&gt;=</span> study.gene.rate.cutoff <span class="sc">|</span></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">fData</span>(object.segment.filtered)<span class="sc">$</span>TargetName <span class="sc">%in%</span> neg.probes, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><strong>Write QC Output File</strong></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Start the QC output excel workbook</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>qc.info.output <span class="ot">&lt;-</span> <span class="fu">createWorkbook</span>()</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the AOI flag info to the output file</span></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a><span class="fu">addWorksheet</span>(qc.info.output, <span class="st">"AOI QC Flags"</span>)</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a><span class="fu">writeData</span>(qc.info.output, <span class="at">sheet =</span> <span class="st">"AOI QC Flags"</span>, qc.output<span class="sc">$</span>segment.flags)</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the probe flag QC info to the output file</span></span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a><span class="fu">addWorksheet</span>(qc.info.output, <span class="st">"Probe QC Flags"</span>)</span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a><span class="fu">writeData</span>(qc.info.output, <span class="at">sheet =</span> <span class="st">"Probe QC Flags"</span>, probe.flags.table)</span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the AOI detection QC info</span></span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a><span class="fu">addWorksheet</span>(qc.info.output, <span class="st">"AOI Detection Rate"</span>)</span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a><span class="fu">writeData</span>(qc.info.output, <span class="at">sheet =</span> <span class="st">"AOI Detection Rate"</span>, segment.detection.summary)</span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the Gene detection QC info to the output file</span></span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a><span class="fu">addWorksheet</span>(qc.info.output, <span class="st">"Gene Detection Rate"</span>)</span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a><span class="fu">writeData</span>(qc.info.output, <span class="at">sheet =</span> <span class="st">"Gene Detection Rate"</span>, gene.detection.summary)</span>
<span id="cb33-19"><a href="#cb33-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-20"><a href="#cb33-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the QC output file</span></span>
<span id="cb33-21"><a href="#cb33-21" aria-hidden="true" tabindex="-1"></a><span class="fu">saveWorkbook</span>(qc.info.output, <span class="fu">paste0</span>(params<span class="sc">$</span>results.folder, params<span class="sc">$</span>run.folder, <span class="st">"QC_info.xlsx"</span>), <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
</section>
<section id="normalization" class="level3">
<h3 class="anchored" data-anchor-id="normalization">Normalization</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>q3.normalization.output <span class="ot">&lt;-</span> <span class="fu">geomxNorm</span>(</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">object =</span> object.gene.filtered, </span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">norm =</span> <span class="st">"q3"</span>)</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>neg.normalization.output <span class="ot">&lt;-</span> <span class="fu">geomxNorm</span>(</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">object =</span> object.gene.filtered, </span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">norm =</span> <span class="st">"neg"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="normalized-reads-vs.-negative-background" class="level4">
<h4 class="anchored" data-anchor-id="normalized-reads-vs.-negative-background">Normalized Reads vs.&nbsp;Negative Background</h4>
<p><a href="#fig-Q3vsBackground">Figure&nbsp;7</a> shows the quartile-3 (Q3) normalization counts compared to negative probes</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>q3.normalization.output<span class="sc">$</span>multi.plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-Q3vsBackground" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/fig-Q3vsBackground-1.png" class="img-fluid figure-img" width="1152"></p>
<figcaption class="figure-caption">Figure&nbsp;7: Q3 Normalized Counts versus Background</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="#fig-NegvsBackground">Figure&nbsp;8</a> shows the negative normalization counts compared to negative probes</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>neg.normalization.output<span class="sc">$</span>multi.plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-NegvsBackground" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/fig-NegvsBackground-1.png" class="img-fluid figure-img" width="1152"></p>
<figcaption class="figure-caption">Figure&nbsp;8: Negative Normalized Counts versus Background</figcaption>
</figure>
</div>
</div>
</div>
<p><strong>Example AOIs</strong></p>
<p><a href="#fig-RawReadBoxPlots">Figure&nbsp;9</a> shows the raw read count boxplots for an example 10 AOIs</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>q3.normalization.output<span class="sc">$</span>boxplot.raw</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-RawReadBoxPlots" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/fig-RawReadBoxPlots-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;9: Raw Counts for an Example 10 AOIs</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="#fig-Q3NormBoxPlots">Figure&nbsp;10</a> shows the Q3 normalized read count boxplots for an example 10 AOIs</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>q3.normalization.output<span class="sc">$</span>boxplot.norm</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-Q3NormBoxPlots" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/fig-Q3NormBoxPlots-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;10: Q3 Normalized Counts for an Example 10 AOIs</figcaption>
</figure>
</div>
</div>
</div>
<p><a href="#fig-NegNormBoxPlots">Figure&nbsp;11</a> shows the negative normalized read count boxplots for an example 10 AOIs</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>neg.normalization.output<span class="sc">$</span>boxplot.norm</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-NegNormBoxPlots" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/fig-NegNormBoxPlots-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption class="figure-caption">Figure&nbsp;11: Negative Normalized Counts for an Example 10 AOIs</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="principal-component-analysis-pca" class="level4">
<h4 class="anchored" data-anchor-id="principal-component-analysis-pca">Principal Component Analysis (PCA)</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># See reference vignette: https://bioconductor.org/packages/release/bioc/vignettes/PCAtools/inst/doc/PCAtools.html#introduction</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the Geomx objects</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>object.q3 <span class="ot">&lt;-</span> q3.normalization.output<span class="sc">$</span>object</span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>object.neg <span class="ot">&lt;-</span> neg.normalization.output<span class="sc">$</span>object</span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Gather the the normalized counts</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>q3.norm.counts.df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(object.q3<span class="sc">@</span>assayData<span class="sc">$</span>q_norm)</span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>neg.norm.counts.df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(object.neg<span class="sc">@</span>assayData<span class="sc">$</span>neg_norm)</span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert counts to log2</span></span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a>q3.log.counts.df <span class="ot">&lt;-</span> q3.norm.counts.df <span class="sc">%&gt;%</span> </span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_all</span>(<span class="sc">~</span> <span class="fu">log2</span>(.)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_all</span>(<span class="sc">~</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">.dcc"</span>, <span class="st">""</span>, .))</span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a>neg.log.counts.df <span class="ot">&lt;-</span> neg.norm.counts.df <span class="sc">%&gt;%</span> </span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_all</span>(<span class="sc">~</span> <span class="fu">log2</span>(.)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename_all</span>(<span class="sc">~</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">.dcc"</span>, <span class="st">""</span>, .))</span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove the negative controls from the log counts</span></span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a>control.probes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"NegProbe-WTX"</span>)</span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a>q3.log.counts.df <span class="ot">&lt;-</span> q3.log.counts.df[<span class="sc">!</span>(<span class="fu">rownames</span>(q3.log.counts.df) <span class="sc">%in%</span> control.probes), ]</span>
<span id="cb40-22"><a href="#cb40-22" aria-hidden="true" tabindex="-1"></a>neg.log.counts.df <span class="ot">&lt;-</span> neg.log.counts.df[<span class="sc">!</span>(<span class="fu">rownames</span>(neg.log.counts.df) <span class="sc">%in%</span> control.probes), ]</span>
<span id="cb40-23"><a href="#cb40-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-24"><a href="#cb40-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the annotation (same for both normalization types)</span></span>
<span id="cb40-25"><a href="#cb40-25" aria-hidden="true" tabindex="-1"></a>annotation <span class="ot">&lt;-</span> <span class="fu">pData</span>(object.q3)</span>
<span id="cb40-26"><a href="#cb40-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-27"><a href="#cb40-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove NTCs</span></span>
<span id="cb40-28"><a href="#cb40-28" aria-hidden="true" tabindex="-1"></a>cleaned.annotation.df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(annotation[annotation<span class="sc">$</span><span class="st">'slide_name'</span> <span class="sc">!=</span> <span class="st">"No Template Control"</span>, ])</span>
<span id="cb40-29"><a href="#cb40-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-30"><a href="#cb40-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Order of rownames of annotation need to match columns of count data</span></span>
<span id="cb40-31"><a href="#cb40-31" aria-hidden="true" tabindex="-1"></a>cleaned.annotation.df <span class="ot">&lt;-</span> cleaned.annotation.df[<span class="fu">order</span>(<span class="fu">rownames</span>(cleaned.annotation.df)), ]</span>
<span id="cb40-32"><a href="#cb40-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-33"><a href="#cb40-33" aria-hidden="true" tabindex="-1"></a>q3.log.counts.df <span class="ot">&lt;-</span> q3.log.counts.df[<span class="fu">order</span>(<span class="fu">colnames</span>(q3.log.counts.df))]</span>
<span id="cb40-34"><a href="#cb40-34" aria-hidden="true" tabindex="-1"></a>neg.log.counts.df <span class="ot">&lt;-</span> neg.log.counts.df[<span class="fu">order</span>(<span class="fu">colnames</span>(neg.log.counts.df))]</span>
<span id="cb40-35"><a href="#cb40-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-36"><a href="#cb40-36" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove .dcc from Sample ID row names</span></span>
<span id="cb40-37"><a href="#cb40-37" aria-hidden="true" tabindex="-1"></a>cleaned.annotation.df <span class="ot">&lt;-</span> cleaned.annotation.df <span class="sc">%&gt;%</span> <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(<span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">.dcc"</span>, <span class="st">""</span>, <span class="fu">rownames</span>(.)))</span>
<span id="cb40-38"><a href="#cb40-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-39"><a href="#cb40-39" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate a PCA table for all samples for both normalization types</span></span>
<span id="cb40-40"><a href="#cb40-40" aria-hidden="true" tabindex="-1"></a>q3.pca.table <span class="ot">&lt;-</span> <span class="fu">pca</span>(q3.log.counts.df, </span>
<span id="cb40-41"><a href="#cb40-41" aria-hidden="true" tabindex="-1"></a>                 <span class="at">metadata =</span> cleaned.annotation.df, </span>
<span id="cb40-42"><a href="#cb40-42" aria-hidden="true" tabindex="-1"></a>                 <span class="at">removeVar =</span> <span class="fl">0.1</span>)</span>
<span id="cb40-43"><a href="#cb40-43" aria-hidden="true" tabindex="-1"></a>neg.pca.table <span class="ot">&lt;-</span> <span class="fu">pca</span>(neg.log.counts.df, </span>
<span id="cb40-44"><a href="#cb40-44" aria-hidden="true" tabindex="-1"></a>                 <span class="at">metadata =</span> cleaned.annotation.df, </span>
<span id="cb40-45"><a href="#cb40-45" aria-hidden="true" tabindex="-1"></a>                 <span class="at">removeVar =</span> <span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="pca-by-segment" class="level4">
<h4 class="anchored" data-anchor-id="pca-by-segment">PCA by Segment</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>q3.pca.plot.segment <span class="ot">&lt;-</span> <span class="fu">biplot</span>(q3.pca.table, </span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">colby =</span> <span class="st">"segment"</span>, </span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendPosition =</span> <span class="st">"right"</span>, </span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendLabSize =</span> <span class="dv">6</span>, </span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendIconSize =</span> <span class="dv">3</span>, </span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">lab =</span> <span class="cn">NULL</span>,</span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">title =</span> <span class="st">"Q3 Normalization"</span>, </span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">subtitle =</span> <span class="st">"NTCs removed"</span>)</span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(q3.pca.plot.segment)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-PCAsegmentQ3" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/fig-PCAsegmentQ3-1.png" class="img-fluid figure-img" width="1152"></p>
<figcaption class="figure-caption">Figure&nbsp;12: PCA colored by Segment for Q3 Normalization</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>neg.pca.plot.segment <span class="ot">&lt;-</span> <span class="fu">biplot</span>(neg.pca.table, </span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">colby =</span> <span class="st">"segment"</span>, </span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendPosition =</span> <span class="st">"right"</span>, </span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendLabSize =</span> <span class="dv">6</span>, </span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendIconSize =</span> <span class="dv">3</span>, </span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">lab =</span> <span class="cn">NULL</span>,</span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">title =</span> <span class="st">"Negative Normalization"</span>, </span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">subtitle =</span> <span class="st">"NTCs removed"</span>)</span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(neg.pca.plot.segment)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-PCAsegmentNeg" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/fig-PCAsegmentNeg-1.png" class="img-fluid figure-img" width="1152"></p>
<figcaption class="figure-caption">Figure&nbsp;13: PCA colored by Segment for Negative Normalization</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="pca-by-region" class="level4">
<h4 class="anchored" data-anchor-id="pca-by-region">PCA by Region</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>q3.pca.plot.region <span class="ot">&lt;-</span> <span class="fu">biplot</span>(q3.pca.table, </span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">colby =</span> <span class="st">"region"</span>, </span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendPosition =</span> <span class="st">"right"</span>, </span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendLabSize =</span> <span class="dv">10</span>, </span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendIconSize =</span> <span class="dv">5</span>, </span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">lab =</span> <span class="cn">NULL</span>,</span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">title =</span> <span class="st">"Q3 Normalization"</span>, </span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">subtitle =</span> <span class="st">"NTCs removed"</span>)</span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(q3.pca.plot.region)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-PCAregionQ3" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/fig-PCAregionQ3-1.png" class="img-fluid figure-img" width="1152"></p>
<figcaption class="figure-caption">Figure&nbsp;14: PCA colored by Region for Q3 Normalization</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>neg.pca.plot.region <span class="ot">&lt;-</span> <span class="fu">biplot</span>(neg.pca.table, </span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">colby =</span> <span class="st">"region"</span>, </span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendPosition =</span> <span class="st">"right"</span>, </span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendLabSize =</span> <span class="dv">10</span>, </span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendIconSize =</span> <span class="dv">5</span>, </span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">lab =</span> <span class="cn">NULL</span>,</span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">title =</span> <span class="st">"Negative Normalization"</span>, </span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">subtitle =</span> <span class="st">"NTCs removed"</span>)</span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(neg.pca.plot.region)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-PCAregionNeg" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/fig-PCAregionNeg-1.png" class="img-fluid figure-img" width="1152"></p>
<figcaption class="figure-caption">Figure&nbsp;15: PCA colored by Region for Negative Normalization</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="pca-by-class" class="level4">
<h4 class="anchored" data-anchor-id="pca-by-class">PCA by Class</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>q3.pca.plot.class <span class="ot">&lt;-</span> <span class="fu">biplot</span>(q3.pca.table, </span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">colby =</span> <span class="st">"class"</span>, </span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendPosition =</span> <span class="st">"right"</span>, </span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendLabSize =</span> <span class="dv">10</span>, </span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendIconSize =</span> <span class="dv">5</span>, </span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">lab =</span> <span class="cn">NULL</span>,</span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">title =</span> <span class="st">"Q3 Normalization"</span>, </span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">subtitle =</span> <span class="st">"NTCs removed"</span>)</span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(q3.pca.plot.class)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-PCAclassQ3" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/fig-PCAclassQ3-1.png" class="img-fluid figure-img" width="1152"></p>
<figcaption class="figure-caption">Figure&nbsp;16: PCA colored by Class for Q3 Normalization</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>neg.pca.plot.class <span class="ot">&lt;-</span> <span class="fu">biplot</span>(neg.pca.table, </span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">colby =</span> <span class="st">"class"</span>, </span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendPosition =</span> <span class="st">"right"</span>, </span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendLabSize =</span> <span class="dv">10</span>, </span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendIconSize =</span> <span class="dv">5</span>, </span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">lab =</span> <span class="cn">NULL</span>,</span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">title =</span> <span class="st">"Negative Normalization"</span>, </span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">subtitle =</span> <span class="st">"NTCs removed"</span>)</span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(neg.pca.plot.class)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-PCAclassNeg" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/fig-PCAclassNeg-1.png" class="img-fluid figure-img" width="1152"></p>
<figcaption class="figure-caption">Figure&nbsp;17: PCA colored by Class for Negative Normalization</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="pca-by-slide" class="level4">
<h4 class="anchored" data-anchor-id="pca-by-slide">PCA by Slide</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>q3.pca.plot.slide <span class="ot">&lt;-</span> <span class="fu">biplot</span>(q3.pca.table, </span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">colby =</span> <span class="st">"slide_name"</span>, </span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendPosition =</span> <span class="st">"right"</span>, </span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendLabSize =</span> <span class="dv">10</span>, </span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendIconSize =</span> <span class="dv">5</span>, </span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">lab =</span> <span class="cn">NULL</span>,</span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">title =</span> <span class="st">"Q3 Normalization"</span>, </span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">subtitle =</span> <span class="st">"NTCs removed"</span>)</span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(q3.pca.plot.slide)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-PCAslideQ3" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/fig-PCAslideQ3-1.png" class="img-fluid figure-img" width="1152"></p>
<figcaption class="figure-caption">Figure&nbsp;18: PCA colored by Slide for Q3 Normalization</figcaption>
</figure>
</div>
</div>
</div>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>neg.pca.plot.slide <span class="ot">&lt;-</span> <span class="fu">biplot</span>(neg.pca.table, </span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">colby =</span> <span class="st">"slide_name"</span>, </span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendPosition =</span> <span class="st">"right"</span>, </span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendLabSize =</span> <span class="dv">10</span>, </span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">legendIconSize =</span> <span class="dv">5</span>, </span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">lab =</span> <span class="cn">NULL</span>,</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">title =</span> <span class="st">"Negative Normalization"</span>, </span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">subtitle =</span> <span class="st">"NTCs removed"</span>)</span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(neg.pca.plot.slide)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-PCAslideNeg" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/fig-PCAslideNeg-1.png" class="img-fluid figure-img" width="1152"></p>
<figcaption class="figure-caption">Figure&nbsp;19: PCA colored by Slide for Negative Normalization</figcaption>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="ma-plots" class="level3">
<h3 class="anchored" data-anchor-id="ma-plots">MA Plots</h3>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up pre and post normalization counts and convert to log</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>pre.norm.counts <span class="ot">&lt;-</span></span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">log</span>(<span class="fu">as.data.frame</span>(q3.normalization.output<span class="sc">$</span>object<span class="sc">@</span>assayData<span class="sc">$</span>exprs), <span class="at">base =</span> <span class="dv">2</span>)</span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a>post.q3norm.counts <span class="ot">&lt;-</span> <span class="fu">log</span>(<span class="fu">as.data.frame</span>(q3.normalization.output<span class="sc">$</span>object<span class="sc">@</span>assayData<span class="sc">$</span>q_norm), <span class="at">base =</span> <span class="dv">2</span>)</span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>post.negnorm.counts <span class="ot">&lt;-</span> <span class="fu">log</span>(<span class="fu">as.data.frame</span>(neg.normalization.output<span class="sc">$</span>object<span class="sc">@</span>assayData<span class="sc">$</span>neg_norm), <span class="at">base =</span> <span class="dv">2</span>)</span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Setup annotation groups for the ratio "M" (log A - log B)</span></span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a>tumor.types <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Biopsy"</span>)</span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a>region.types <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"tumor"</span>, <span class="st">"vessel"</span>)</span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a>source.types <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Skin"</span>)</span>
<span id="cb49-14"><a href="#cb49-14" aria-hidden="true" tabindex="-1"></a>segment.types <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"full_ROI"</span>)</span>
<span id="cb49-15"><a href="#cb49-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-16"><a href="#cb49-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the name of the contrast</span></span>
<span id="cb49-17"><a href="#cb49-17" aria-hidden="true" tabindex="-1"></a>contrast.name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(tumor.types[[<span class="dv">1</span>]], </span>
<span id="cb49-18"><a href="#cb49-18" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"_"</span>,</span>
<span id="cb49-19"><a href="#cb49-19" aria-hidden="true" tabindex="-1"></a>                        source.types[[<span class="dv">1</span>]], </span>
<span id="cb49-20"><a href="#cb49-20" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"_"</span>, </span>
<span id="cb49-21"><a href="#cb49-21" aria-hidden="true" tabindex="-1"></a>                        segment.types[[<span class="dv">1</span>]], </span>
<span id="cb49-22"><a href="#cb49-22" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"_"</span>, </span>
<span id="cb49-23"><a href="#cb49-23" aria-hidden="true" tabindex="-1"></a>                        region.types[[<span class="dv">1</span>]], </span>
<span id="cb49-24"><a href="#cb49-24" aria-hidden="true" tabindex="-1"></a>                        <span class="st">"_"</span>, </span>
<span id="cb49-25"><a href="#cb49-25" aria-hidden="true" tabindex="-1"></a>                        region.types[[<span class="dv">2</span>]])</span>
<span id="cb49-26"><a href="#cb49-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-27"><a href="#cb49-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up the annotations and raw counts for the MA plots</span></span>
<span id="cb49-28"><a href="#cb49-28" aria-hidden="true" tabindex="-1"></a>contrast.field <span class="ot">&lt;-</span> <span class="st">"region"</span></span>
<span id="cb49-29"><a href="#cb49-29" aria-hidden="true" tabindex="-1"></a>condition.label <span class="ot">&lt;-</span> <span class="st">"tumor"</span></span>
<span id="cb49-30"><a href="#cb49-30" aria-hidden="true" tabindex="-1"></a>reference.label <span class="ot">&lt;-</span> <span class="st">"vessel"</span></span>
<span id="cb49-31"><a href="#cb49-31" aria-hidden="true" tabindex="-1"></a>raw.log.counts <span class="ot">&lt;-</span> pre.norm.counts</span>
<span id="cb49-32"><a href="#cb49-32" aria-hidden="true" tabindex="-1"></a>annotation.MA <span class="ot">&lt;-</span> annotation</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="q3-normalization" class="level4">
<h4 class="anchored" data-anchor-id="q3-normalization">Q3 Normalization</h4>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># MA plots for two annotation groups to evaluate negative normalization</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>log.counts <span class="ot">&lt;-</span> post.q3norm.counts</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>MA.plots.q3 <span class="ot">&lt;-</span> <span class="fu">make_MA</span>(<span class="at">contrast.field =</span> contrast.field, </span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>                       <span class="at">condition.label =</span> condition.label, </span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>                       <span class="at">reference.label =</span> reference.label, </span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">log.counts =</span> log.counts, </span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a>                       <span class="at">raw.log.counts =</span> raw.log.counts, </span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">annotation =</span> annotation.MA)</span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a><span class="fu">grid.draw</span>(MA.plots.q3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/MA%20Plot%20Q3-1.png" class="img-fluid" width="1152"></p>
</div>
</div>
</section>
<section id="negative-normalization" class="level4">
<h4 class="anchored" data-anchor-id="negative-normalization">Negative Normalization</h4>
<p>grid.draw(MA.plots.neg)</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="co"># MA plots for two annotation groups to evaluate negative normalization</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>log.counts <span class="ot">&lt;-</span> post.negnorm.counts</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>MA.plots.neg <span class="ot">&lt;-</span> <span class="fu">make_MA</span>(<span class="at">contrast.field =</span> contrast.field, </span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">condition.label =</span> condition.label, </span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">reference.label =</span> reference.label, </span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">log.counts =</span> log.counts, </span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">raw.log.counts =</span> raw.log.counts, </span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">annotation =</span> annotation.MA)</span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb51-11"><a href="#cb51-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb51-12"><a href="#cb51-12" aria-hidden="true" tabindex="-1"></a><span class="fu">grid.draw</span>(MA.plots.neg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="DSP_Analysis_QC_Report_files/figure-html/MA%20Plot%20Neg-1.png" class="img-fluid" width="1152"></p>
</div>
</div>
</section>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>